src/storage.cpp

```diff
@@ -1692,6 +1693,9 @@ namespace libtorrent
 	{
 		TORRENT_ASSERT(m_files.piece_length() > 0);
 
+		if (checked_fastresume) return check_no_fastresume(ec);
```

Comment:
the problem I would be worried about (apart from this trying to treat the symptom rather than the illness) is that there may be legitimate cases of calling async_check_fastresume() twice. For instance when force-rechecking a torrent, or when promoting a "magnet link" into a proper torrent (either by it receiving metadata or by adding a .torrent file that match the info-hash)